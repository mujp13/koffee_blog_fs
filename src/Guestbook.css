.Login__form {
  padding: 20px;
}

.button {
  width: 100px;
  float: right;
}

#comment {
  clear: right;
  width: 99.4%;
  height: 100px;
}

#person_name {
  padding: 1%;
}
