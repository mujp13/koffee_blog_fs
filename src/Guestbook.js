import React, { Component } from 'react';

export default class Guestbook extends Component {
  render() {
    return (
      <form className="Login__form">
        <div>
          <label htmlFor="person_name" />
          <input type="text" name="person_name" id="person_name" placeholder="Name" required />
        </div>
        <div>
          <label htmlFor="comment" />
          <textarea name="comment" id="comment" placeholder="Put your precious comments here" required></textarea>
        </div>
        <div className="send__button">
          <button type="submit">Send</button>
        </div>
      </form>
    );
  }
}
