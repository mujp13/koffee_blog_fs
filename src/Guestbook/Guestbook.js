import React, { Component } from 'react';
import KoffeeBlogContext from './KoffeeBlogContext/KoffeeBlogContext';
import './Guestbook.css';
import GuestPost from './GuestPost/GuestPost';

export default class Guestbook extends Component {
  static contextType = KoffeeBlogContext;

  state = {
    error: null,
  };

  handleSubmit = (e) => {
    e.preventDefault();
    // get the form fields from the event
    const { name, comment } = e.target;
    const guestbook = {
      name: name.value,
      comment: comment.value,
    };

    this.setState({ error: null });

    fetch('https://friendly-donair-37840.herokuapp.com/api/guestbook', {
      method: 'POST',
      body: JSON.stringify(guestbook),
      headers: {
        'content-type': 'application/json',
      },
    })
      .then((res) => {
        if (!res.ok) {
          // get the error message from the response,
          return res.json().then((error) => {
            // then throw it
            throw error;
          });
        }
        return res.json();
      })
      .then((data) => {
        console.log(data);
        this.context.addGuestbook(data);
        this.props.history.push('/guestbook');
      })
      .catch((error) => {
        this.setState({ error });
      });
  };

  render() {
    const { error } = this.state;

    return (
      <>
        <h2>Thanks for visiting! Please come again!</h2>
        <form className="Login__form" onSubmit={this.handleSubmit}>
          <div className="addGuestbook" role="alert">
            {error && <p>{error.messsage}</p>}
          </div>
          <div>
            <label htmlFor="name" />
            <input type="text" name="name" id="name" placeholder="Name" required />
          </div>
          <div>
            <label htmlFor="comment" />
            <textarea name="comment" id="comment" placeholder="Put your precious comments here" required></textarea>
          </div>
          <div className="send__button">
            <button className="button" type="submit">
              Send
            </button>
          </div>
        </form>
        <section className="guest-comment">
          {this.context.guestbooks.map((post) => (
            <GuestPost name={post.name} comment={post.comment} date={post.date_published} />
          ))}
        </section>
      </>
    );
  }
}
