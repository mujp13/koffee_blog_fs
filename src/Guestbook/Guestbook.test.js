import React from 'react';
import ReactDOM from 'react-dom';
import Guestbook from './Guestbook';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <Guestbook />
    </MemoryRouter>,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
});

//snapshot
it('renders correctly', () => {
  const tree = renderer.create(<Guestbook />).toJSON();
  expect(tree).toMatchSnapshot();
});
