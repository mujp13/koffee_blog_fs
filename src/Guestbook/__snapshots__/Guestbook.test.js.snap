// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h2>
    Thanks for visiting! Please come again!
  </h2>,
  <form
    className="Login__form"
    onSubmit={[Function]}
  >
    <div
      className="addGuestbook"
      role="alert"
    />
    <div>
      <label
        htmlFor="name"
      />
      <input
        id="name"
        name="name"
        placeholder="Name"
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        htmlFor="comment"
      />
      <textarea
        id="comment"
        name="comment"
        placeholder="Put your precious comments here"
        required={true}
      />
    </div>
    <div
      className="send__button"
    >
      <button
        className="button"
        type="submit"
      >
        Send
      </button>
    </div>
  </form>,
  <section
    className="guest-comment"
  />,
]
`;
